File: core/causal_intelligence/semantic_execution_stability_horizon_engine.py
from __future__ import annotations

from typing import Any, Dict


def forecast_stability_horizon(
    runtime_ir: Dict[str, Any],
) -> Dict[str, Any]:
    instability = runtime_ir.get("instability_forecast", {})
    if isinstance(instability, dict):
        level = instability.get("instability_forecast", "low")
    else:
        level = "low"
    horizon = (
        "short"
        if level == "high"
        else "long"
    )
    return {
        "stability_horizon": horizon,
        "instability_level": level,
    }

File: src/causal_intelligence/semanticExecutionStabilityHorizonEngine.ts
/**
 * Converted from Python: core/causal_intelligence/semantic_execution_stability_horizon_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function forecastStabilityHorizon(runtime_ir: any): any {
  var instability: any = py.get(runtime_ir, "instability_forecast", {});
  if (((instability !== null && typeof instability === "object" && !Array.isArray(instability) && !(instability instanceof Set) && !(instability instanceof Map)))) {
    var level: any = py.get(instability, "instability_forecast", "low");
  } else {
    level = "low";
  }
  var horizon: any = (py.eq(level, "high") ? "short" : "long");
  return {"stability_horizon": horizon, "instability_level": level};
}
